### PR TITLE
[Ops] Converted from cookie setting to jwt token response

### PR DESCRIPTION
**Pull request by Weiyuan-Lane**
**Objectives**
Switched authentication endpoint from returning a set cookie (session) header to a token in the response. 

This token (jwt) is tightly coupled with the other endpoints that can be built on this platform, where the user from the token can be retrieved from these endpoints. More options have been added to allow for `optional` and `required` authentication, whereby `optional` embeds `req.user` if it is able to find a valid token, while `required` throws a forbidden error if it is not able locate a user entity from the token.

Frontend clients are to use send it as a bearer token, whereby this app will extract and perform its operations from there automatically

**Others**
- Also remove all HTML outputs
- Ensure all returns from login flow ends as a JSON output for client decode and use (e.g. display of error message)
- Added a fallback 404 :D

### CODE REVIEW

File: src/auth/auth.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+const path = require('path');
+const passport = require(path.resolve('src/auth/passport'));
+const jwt = require('jsonwebtoken');
+
+// login drives the workflow for a login strategy to be used and ensures
+// that a valid token is returned in the response
+exports.login = function (req, res, next) {
```
Comment (briantjt) [inline src/auth/auth.js]: Why wrap passport.authenticate in a function that takes the arguments and just passes it into passport.authenticate? Wouldn't it be the same as just

`
exports.login = passport.authenticate('local', { session: false }, function (err, user) {
    if (err) {
      return res.status(500).json({ message: `500 ${err.message}` });
    }
    if (!user) {
      return res.status(401).json({ message: '401 credentials are invalid' });
    }

    // Replacement for serialising the returned token
    const token = jwt.sign({ userId: user.id }, process.env.EXPRESS_SESSION_SECRET);
    res.json({ token });
  `

Comment (Weiyuan-Lane) [reply src/auth/auth.js]: no `res` :/

**Comment by Weiyuan-Lane** [comment]
Additional note for testing and integrating. Please insert the following header for passing the token value to the endpoint

```
Authorization: Bearer ${tokenValue}
```

**Comment by briantjt** [comment]
@Weiyuan-Lane good to merge?

**Pull request merged**